062-topic-replication-factor-change.md

```diff
@@ -0,0 +1,254 @@
+
+During the whole process, the `KafkaTopic` remains in a `Ready` state, so we use an additional `Warning` condition to communicate the replication factor change status.
+
+<figure>
+    <img src="images/062-topic-replication-factor-change-upd.png" height=120>
+    <figcaption><small>
+        Fig 1. KafkaTopic status updates (happy path).
+    </small></figcaption>
+</figure>
+
+<br/>
+<br/>
+
+To request a replication factor change, the Topic Operator needs to authenticate and send HTTP requests to the Cruise Control's REST API.
+We introduce the following environment variables, which will be set automatically by the Cluster Operator, or manually by the user in case of standalone deployment.
+
+| Name                                   | Type    | Default | Description                                                   |
+|:---------------------------------------|---------|---------|---------------------------------------------------------------|
+| STRIMZI_CRUISE_CONTROL_ENABLED         | Boolean | false   | Whether Cruise Control integration is enabled                 |
+| STRIMZI_CRUISE_CONTROL_NAMESPACE       | String  | ""      | Namespace where Cruise Control is running                     |
+| STRIMZI_CRUISE_CONTROL_HOSTNAME        | String  | ""      | Cruise Control hostname                                       |
+| STRIMZI_CRUISE_CONTROL_PORT            | Integer | 9090    | Cruise Control port                                           |
+| STRIMZI_CRUISE_CONTROL_SSL_ENABLED     | Boolean | false   | Whether SSL encryption is enabled                             |
+| STRIMZI_CRUISE_CONTROL_AUTH_ENABLED    | Boolean | false   | Whether Basic authentication is enabled                       |
+| STRIMZI_CRUISE_CONTROL_RACK_ENABLED    | Boolean | false   | Whether rack awareness is enabled in the target Kafka cluster |
+| STRIMZI_CRUISE_CONTROL_SECRET_NAME     | String  | ""      | Secret name containing the truststore                         |
+| STRIMZI_CRUISE_CONTROL_API_SECRET_NAME | String  | ""      | Secret name containing the API credentials                    |
+
+For the standalone deployment, the Topic Operator expects the following secrets:
+
+- A secret with `cruise-control.p12` key containing the truststore in PKCS12 format, and `cruise-control.password` key containing the truststore password.
+- A secret with `cruise-control.apiToAdminUsername` key containing the API username with `ADMIN` role, and `cruise-control.apiToAdminPassword` key containing the user password.
+
+Cruise Control's topic configuration endpoint can be used to request a replication factor change.
+Once accepted, the replication factor change is put into a task queue and executed asynchronously.
+A task is first created in `Active` state, then moved to `InExecution`, and finally to `Completed` or `CompletedWithError`.
+Only a single task can be `InExecution` at any given time, which includes both topic configuration and rebalance tasks.
```

Comment:
Are we talking about Cruise Control tasks ... or TO tasks as something you are adding here in the proposal? Can you make it clearer, please?

Reply:
CC, rephrased to make it clear.